odl/tomo/data/uncompr_bin.py

```diff
@@ -0,0 +1,472 @@
+        """
+        # Read all fields except data
+        header = {}
+        for field in self.header_fields:
+            # Get all the values from the dictionary
+            name = field['name']
+            if name == 'data':
+                continue
+            entry = {'description': field.get('description', '')}
+            offset_bytes = field['offset_bytes']
+            size_bytes = field.get('size_bytes', None)
+            dtype = field['dtype']
+            shape = field.get('dshape', None)
+            if shape is None:
+                shape = -1  # Causes reshape to 1d or no-op
+
+            bytes_per_elem = dtype.itemsize
+
+            if size_bytes is None:
+                # Default if 'size_bytes' is omitted
+                num_elems = 1
+            else:
+                num_elems = size_bytes / bytes_per_elem
+
+            if num_elems != int(num_elems):
+                raise RuntimeError(
+                    "field '{}': `size_bytes` {} and `dtype.itemsize` {} "
+                    " result in non-integer number of elements"
+                    "".format(name, size_bytes, bytes_per_elem))
+
+            # Create format string for struct module to unpack the binary
+            # data
+            fmt = str(int(num_elems)) + DTYPE_MAP_NPY2STRUCT[dtype]
+            if struct.calcsize(fmt) != size_bytes:
+                raise RuntimeError(
+                    "field '{}': format '{}' has results in {} bytes, but "
+                    "`size_bytes` is {}"
+                    "".format(name, fmt, struct.calcsize(fmt), size_bytes))
+
+            self.file.seek(offset_bytes)
+            packed_value = self.file.read(size_bytes)
+            value = np.array(struct.unpack_from(fmt, packed_value),
+                             dtype=dtype)
+
+            if dtype == np.dtype('S1'):
+                entry['value'] = ''.join(value.astype(str)).ljust(size_bytes)
+            else:
+                entry['value'] = value.reshape(shape)
+            header[name] = entry
+
+        # Store information gained from the header
+        self.header = header
+        self._set_attrs_from_header()
+
+        return header
+
+    def _set_attrs_from_header(self):
+        """Abstract method for setting attributes from ``self.header``.
+
+        The minimum attributes to set by this method are:
+
+            - ``data_shape`` : Shape of the (full) data.
+            - ``data_dtype`` : Data type of the data.
+
+        Subclasses must override this method. They may define additional
+        attributes to set.
+        """
+        raise NotImplementedError
+
+    def read_data(self, dstart=None, dend=None):
+        """Read the data from the reader's file and store if desired.
+
+        Parameters
+        ----------
+        dstart : int, optional
+            Offset in bytes of the data field. By default, it is equal
+            to ``header_size``. Negative values are added to the file
```

Comment:
IMO, that version says much less that the text right now - "python style" is not a well-defined notion.